Project setup

PrangonGhose opened the pull request:
### Functionalities added to this PR
- Use of Gitflow
- Use of npm project with webpack
- Less styling used only to satisfy the wireframe
- Use of ES6 modules
- All functionalities implemented

PrangonGhose committed:
Update package.json

PrangonGhose committed:
Update package-lock.json

PrangonGhose committed:
Update webpack.config.js

PrangonGhose committed:
Add index js created by webpack

PrangonGhose committed:
Add runtime js created by webpack

PrangonGhose committed:
Add main html created by webpack

PrangonGhose committed:
Add template HTML

PrangonGhose committed:
Add Primary JS file

PrangonGhose committed:
Add dom object

PrangonGhose committed:
Add functionality to add from form

PrangonGhose committed:
Add leader object instance

PrangonGhose committed:
Add styling to the HTML

PrangonGhose committed:
Add accessing localStorage functionality

PrangonGhose committed:
Update README.md

anagudelogu commented on README.md:
```diff
@@ -1 +1,195 @@
+
+# 📖 Project Name
+
+#### To Do List Project
```
- [ ] Please consider changing your project title to the correct one! 🤓 🛠️

Olamarx replied:
> * [ ]  Please consider changing your project title to the correct one! 🤓 🛠️

- [ ] Nice one fixing the link to the project as required by the previous reviewer, however, you are yet to fix this aspect as the project title still reads To Do List Project, this is a Leaderboard project. Kindly fix it 😃

anagudelogu commented on README.md:
````diff
@@ -1 +1,195 @@
+```
+### Setup
+
+Clone the repository using [this link](https://github.com/PrangonGhose/To-do-list.git)
````
- [ ] Also update this link, as it is pointing to the To-do list repo instead of this one! 🤓

anagudelogu commented on README.md:
```diff
@@ -1 +1,195 @@
+
+Contributions, issues, and feature requests are welcome! Add suggestions by opening new issues.
+
+Feel free to check the [issues page](../../issues/).
```
- [ ] **[OPTIONAL]:** I suggest you add a link to the issues page of this repo, so people that want to contribute can have an easy way to create new issues! 🤓 🛠️

anagudelogu commented on src/formAdd.js:
```diff
@@ -0,0 +1,44 @@
+const leaderName = document.querySelector('.leader-name');
+const leaderScore = document.querySelector('.leader-score');
+export const uiObject = new UI(); // object to be displayed in the DOM
+export let leadersArray = Storage.getStorage(); // eslint-disable-line import/no-mutable-exports
```
- [ ] **[OPTIONAL]:** Although this was not required in this milestone, please remember to avoid disabling the linters, they ensure a consistent format and best practices across your source files! 🤓 🛠️

anagudelogu reviewed (changes requested):
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @PrangonGhose, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 

<img src="https://media2.giphy.com/media/3ohhwi25ISXC7Z4tMs/giphy-downsized-medium.gif"/>

## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 
✅ Great job implementing JS functionality! 🤖  

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

PrangonGhose committed:
Update README.md

PrangonGhose committed:
Update in webpack dist

PrangonGhose committed:
Update formAdd in src to avoid eslint

PrangonGhose committed:
Update index.js in src

PrangonGhose committed:
Update to avoid linters

PrangonGhose commented:
### Changes made after first review
- Changed project name
- Added link to the issues page
- Changed proper link to the repo
- Removed disabling eslint rule

Olamarx reviewed (changes requested):
Hi @PrangonGhose, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
<img src="https://media0.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif"/>

 ## Highlights
- Linters passed ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

PrangonGhose committed:
Update README.md

danmainah reviewed (approved):
Hi @PrangonGhose , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️  Good design

## Optional suggestions

- [ ] [optional] Since this is a javascript project the project requires you to use GitHub flow at this point in the curriculum instead of the gitflow used, kindly consider changing it.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
